### PR TITLE
docs: update sizes="any" apply strategy to icons

### LINKED ISSUE #69553: Favicon.ico link not using `sizes="any"`
### Link to the code that reproduces this issue

https://github.com/seamuslowry/real-true-science-facts/tree/bug/no_any

### To Reproduce

1. run `npm run build`
2. note that the generated `out/index.html` file has `<link rel="icon" href="/favicon.ico" type="image/x-icon" sizes="16x16"/>`

### Current vs. Expected behavior

According to the [documentation](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/app-icons#favicon), generated `head` data from `favicon.ico` should use `sizes="any"`. In this case, it uses `sizes="16x16"`. And the sizes field generated is based on the sizes of the images in `favicon.ico`

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP PREEMPT_DYNAMIC Debian 6.1.106-3 (2024-08-26)
  Available memory (MB): 31740
  Available CPU cores: 20
Binaries:
  Node: 20.12.2
  npm: 10.5.0
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.2.7
  eslint-config-next: 14.2.2
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.5
Next.js Config:
  output: export
```

### Which area(s) are affected? (Select all that apply)

Documentation, Metadata, Output (export/standalone)

### Which stage(s) are affected? (Select all that apply)

next build (local), next start (local), Other (Deployed)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by devjiwonchoi**
### Why?

When adding `app/favicon.ico`, the `sizes` property were set to `"any"` to [avoid a browser bug](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs) where an `.ico` icon is favored over `.svg`. However, we changed the behavior to set the actual size at https://github.com/vercel/next.js/pull/53343 but the doc wasn't updated together.

### How?

Removed the previous explanation and updated that`sizes="any"` may be applied if the extension is `.svg` or the size is not specified.

Closes #69553
Closes NDX-284

### CODE REVIEW

**Comment by devjiwonchoi** [comment]
* **#69975** <a href="https://app.graphite.dev/github/pr/vercel/next.js/69975?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @devjiwonchoi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/02-app/02-api-reference/02-file-conventions/01-metadata/app-icons.mdx [hunk 1]
```diff
@@ -64,8 +64,8 @@ Add an `apple-icon.(jpg|jpeg|png)` image file.
 > - You can set multiple icons by adding a number suffix to the file name. For example, `icon1.png`, `icon2.png`, etc. Numbered files will sort lexically.
 > - Favicons can only be set in the root `/app` segment. If you need more granularity, you can use [`icon`](#icon).
 > - The appropriate `<link>` tags and attributes such as `rel`, `href`, `type`, and `sizes` are determined by the icon type and metadata of the evaluated file.
->   - For example, a 32 by 32px `.png` file will have `type="image/png"` and `sizes="32x32"` attributes.
-> - `sizes="any"` is added to `favicon.ico` output to [avoid a browser bug](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs) where an `.ico` icon is favored over `.svg`.
+> - For example, a 32 by 32px `.png` file will have `type="image/png"` and `sizes="32x32"` attributes.
+> - `sizes="any"` may be applied if the extension is `.svg` or the size is not specified.
```
Comment (huozhi) [inline docs/02-app/02-api-reference/02-file-conventions/01-metadata/app-icons.mdx:68]:
```suggestion
> - `sizes="any"` is only added to `favicon.ico` when the image size of the file is not determined. More details in this [favicon handbook](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs)
```

**Review by huozhi — CHANGES_REQUESTED** [review]

File: docs/02-app/02-api-reference/02-file-conventions/01-metadata/app-icons.mdx [hunk 2]
```diff
@@ -64,8 +64,8 @@ Add an `apple-icon.(jpg|jpeg|png)` image file.
 > - You can set multiple icons by adding a number suffix to the file name. For example, `icon1.png`, `icon2.png`, etc. Numbered files will sort lexically.
 > - Favicons can only be set in the root `/app` segment. If you need more granularity, you can use [`icon`](#icon).
 > - The appropriate `<link>` tags and attributes such as `rel`, `href`, `type`, and `sizes` are determined by the icon type and metadata of the evaluated file.
->   - For example, a 32 by 32px `.png` file will have `type="image/png"` and `sizes="32x32"` attributes.
-> - `sizes="any"` is added to `favicon.ico` output to [avoid a browser bug](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs) where an `.ico` icon is favored over `.svg`.
+> - For example, a 32 by 32px `.png` file will have `type="image/png"` and `sizes="32x32"` attributes.
+> - `sizes="any"` is only added to `favicon.ico` when the image size of the file is not determined. More details in this [favicon handbook](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs)
```
Comment (huozhi) [inline docs/02-app/02-api-reference/02-file-conventions/01-metadata/app-icons.mdx:68]:
```suggestion
> - `sizes="any"` be added to icons when the extension is ".svg" or the image size of the file is not determined. More details in this [favicon handbook](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs)
```

File: docs/02-app/02-api-reference/02-file-conventions/01-metadata/app-icons.mdx [hunk 3]
```diff
@@ -64,8 +64,8 @@ Add an `apple-icon.(jpg|jpeg|png)` image file.
 > - You can set multiple icons by adding a number suffix to the file name. For example, `icon1.png`, `icon2.png`, etc. Numbered files will sort lexically.
 > - Favicons can only be set in the root `/app` segment. If you need more granularity, you can use [`icon`](#icon).
 > - The appropriate `<link>` tags and attributes such as `rel`, `href`, `type`, and `sizes` are determined by the icon type and metadata of the evaluated file.
->   - For example, a 32 by 32px `.png` file will have `type="image/png"` and `sizes="32x32"` attributes.
-> - `sizes="any"` is added to `favicon.ico` output to [avoid a browser bug](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs) where an `.ico` icon is favored over `.svg`.
+> - For example, a 32 by 32px `.png` file will have `type="image/png"` and `sizes="32x32"` attributes.
+> - `sizes="any"` be added to icons when the extension is ".svg" or the image size of the file is not determined. More details in this [favicon handbook](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs)
```
Comment (devjiwonchoi) [inline docs/02-app/02-api-reference/02-file-conventions/01-metadata/app-icons.mdx:68]:
```suggestion
> - `sizes="any"` be added to icons when the extension is `.svg` or the image size of the file is not determined. More details in this [favicon handbook](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs).
```

File: docs/02-app/02-api-reference/02-file-conventions/01-metadata/app-icons.mdx [hunk 4]
```diff
@@ -64,8 +64,8 @@ Add an `apple-icon.(jpg|jpeg|png)` image file.
 > - You can set multiple icons by adding a number suffix to the file name. For example, `icon1.png`, `icon2.png`, etc. Numbered files will sort lexically.
 > - Favicons can only be set in the root `/app` segment. If you need more granularity, you can use [`icon`](#icon).
 > - The appropriate `<link>` tags and attributes such as `rel`, `href`, `type`, and `sizes` are determined by the icon type and metadata of the evaluated file.
->   - For example, a 32 by 32px `.png` file will have `type="image/png"` and `sizes="32x32"` attributes.
-> - `sizes="any"` is added to `favicon.ico` output to [avoid a browser bug](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs) where an `.ico` icon is favored over `.svg`.
+> - For example, a 32 by 32px `.png` file will have `type="image/png"` and `sizes="32x32"` attributes.
+> - `sizes="any"` be added to icons when the extension is `.svg` or the image size of the file is not determined. More details in this [favicon handbook](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs).
```
Comment (huozhi) [inline docs/02-app/02-api-reference/02-file-conventions/01-metadata/app-icons.mdx:68]:
```suggestion
> - `sizes="any"` is added to icons when the extension is `.svg` or the image size of the file is not determined. More details in this [favicon handbook](https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs).
```

**Review by huozhi — APPROVED** [review]

**Review by samcx — APPROVED** [review]

**Pull request merged**